testsuite/gateway/__init__.py

```diff
@@ -1,13 +1,15 @@
 """Classes related to Gateways"""
+import abc
+import enum
 from abc import ABC, abstractmethod
 from dataclasses import dataclass
-from typing import Any, Optional, TYPE_CHECKING
+from typing import Any, Optional, TYPE_CHECKING, Literal
 
 from httpx import Client
 
 from testsuite.certificates import Certificate
 from testsuite.lifecycle import LifecycleObject
-from testsuite.utils import asdict
+from testsuite.utils import asdict, _asdict_recurse
```

Comment:
In this situation I would consider making `_asdict_recurse` a public function as it seems that more and more code depends on it.

Reply:
Removed in last PR update